packages/components/src/utils/getLocation.ts

```diff
@@ -0,0 +1,27 @@
+export interface Coordinates {
+  latitude: number;
+  longitude: number;
+}
+
+export interface Location extends Coordinates {
+  address: string;
+}
+
+const getLocation = async (
+  addressOrLocation: string | Coordinates,
+  geocoder: google.maps.Geocoder
+): Promise<Location> => {
+  const data = await geocoder!.geocode({
+    ...(typeof addressOrLocation === 'string'
+      ? { address: addressOrLocation }
+      : { location: { lat: addressOrLocation.latitude, lng: addressOrLocation.longitude } })
+  });
+
+  const latitude = data.results[0].geometry.location.lat();
```

Comment:
If geocode fails will these error?

Reply:
@jknapp25 I'll need to think about this one, good catch. IMO we can release since this is how it's handled now, but we should track this with a task to show a error handler somehow

Reply:
maybe optional chaining is a quick safety check?